File: Kattis/bookingaroom.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int r, n;
    map <int, bool> m;
    cin >> r >> n;
    for(int i = 0; i < n; i++) {
        int t;
        cin >> t;
        m[t]= true;
    }
    for(int i = 1; i<r; i++) {
        if(m.contains(i) == false) {
            cout << i << endl;
            return 1;
        }
    }
    cout << "too late" << endl;
}

File: Kattis/bookingaroom.py
r, n = map(int, input().split())
d = {}
for i in range(n):
    d[int(input())] = True

f = False
for i in range(1, r + 1):
    if not i in d:
        print(i)
        f = True
        break
if f == False:
    print("too late")